Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -1519,6 +1626,125 @@ namespace gdjs {
       );
     };
 
+    const onLobbyQuickJoinFinished = (runtimeScene: gdjs.RuntimeScene) => {
+      _isJoiningOrStartingAGame = false;
+      gdjs.multiplayerComponents.displayLoader(runtimeScene, false);
+    };
+
+    const doQuickJoinLobby = async (
+      runtimeScene: gdjs.RuntimeScene,
+      displayLoader: boolean
+    ) => {
+      if (_isJoiningOrStartingAGame) return;
+
+      const _gameId = gdjs.projectData.properties.projectUuid;
+      if (!_gameId) {
+        handleLobbiesError(
+          runtimeScene,
+          'The game ID is missing, the quick join lobby action cannot continue.'
+        );
+        return;
+      }
+
+      _isJoiningOrStartingAGame = true;
+      if (displayLoader) {
+        gdjs.multiplayerComponents.displayLoader(runtimeScene, true);
+      }
+
+      const quickJoinLobbyRelativeUrl = `/play/game/${_gameId}/public-lobby/action/quick-join`;
+
+      try {
+        const quickJoinLobbyResponse: QuickJoinLobbyResponse = await gdjs.evtTools.network.retryIfFailed(
+          { times: 2 },
+          () =>
+            // TODO: should the supported compression methods be sent to the endpoint?
+            fetchAsPlayer({
+              relativeUrl: quickJoinLobbyRelativeUrl,
+              method: 'POST',
+              dev: isUsingGDevelopDevelopmentEnvironment,
+              additionalQueryStringParameters: {
+                isPreview: runtimeScene.getGame().isPreview().toString(),
+                gameVersion: runtimeScene.getGame().getGameData().properties
+                  .version,
+              },
+            })
+        );
+
+        if (quickJoinLobbyResponse.status === 'full') {
+          gdjs.multiplayerComponents.showNotification({
+            runtimeScene,
+            content: 'All lobbies are full.',
+            type: 'error',
+          });
+          return;
+        }
+        if (quickJoinLobbyResponse.status === 'not-enough-players') {
+          // TODO: Should this notification be displayed? The creator might not want it displayed
```

Comment:
I feel it should not be displayed indeed. or maybe the action "join quick lobby" should have a parameter `openLobbiesWindowIfNoneAvailable` that can be set to false if needed, but true by default

Reply:
@4ian Do you concur?

Reply:
Agree that probably nothing shown and we let the creator handle the result?
I fear that this is the kind of notification that is nice to follow what's going on, but that will not be liked by most game creators.